File: CS/Form1.cs
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
// ...

namespace CreateSubreport {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        public XtraReport CreateCompositeReport(XtraReport detailReport) {
            // Create a report.
            XtraReport mainReport = new XtraReport();

            // Create a Subreport.
            XRSubreport subreport = new XRSubreport();

            // Create a detail band and add it to the main report.
            DetailBand detailBand = new DetailBand();
            mainReport.Bands.Add(detailBand);

            // Set the subreport's location.
            subreport.Location = new Point(110, 100);

            // Specify a detail report as a report source for the subreport.
            subreport.ReportSource = detailReport;

            // Add the subreport to the detail band.
            detailBand.Controls.Add(subreport);

            return mainReport;
        }

        private void button1_Click(object sender, System.EventArgs e) {
            XtraReport report = CreateCompositeReport(new XtraReport1());
            ReportPrintTool printTool = new ReportPrintTool(report);
            printTool.ShowPreviewDialog();
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
' ...

Namespace CreateSubreport
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Public Function CreateCompositeReport(ByVal detailReport As XtraReport) As XtraReport
			' Create a report.
			Dim mainReport As New XtraReport()

			' Create a Subreport.
			Dim subreport As New XRSubreport()

			' Create a detail band and add it to the main report.
			Dim detailBand As New DetailBand()
			mainReport.Bands.Add(detailBand)

			' Set the subreport's location.
			subreport.Location = New Point(110, 100)

			' Specify a detail report as a report source for the subreport.
			subreport.ReportSource = detailReport

			' Add the subreport to the detail band.
			detailBand.Controls.Add(subreport)

			Return mainReport
		End Function

		Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
			Dim report As XtraReport = CreateCompositeReport(New XtraReport1())
			Dim printTool As New ReportPrintTool(report)
			printTool.ShowPreviewDialog()
		End Sub
	End Class
End Namespace